packages/rs-drive-abci/src/validation/state_transition/data_contract_create.rs

```diff
@@ -0,0 +1,129 @@
+use drive::grovedb::TransactionArg;
+use drive::drive::Drive;
+
+use crate::error::Error;
+use crate::platform::PlatformRef;
+use crate::rpc::core::CoreRPCLike;
+use crate::validation::state_transition::key_validation::validate_state_transition_identity_signature;
+use crate::validation::state_transition::StateTransitionValidation;
+
+use super::common::validate_schema;
+
+impl StateTransitionValidation for DataContractCreateTransition {
+    fn validate_structure(
+        &self,
+        _drive: &Drive,
+        _tx: TransactionArg,
+    ) -> Result<SimpleConsensusValidationResult, Error> {
+        let result = validate_schema(&DATA_CONTRACT_CREATE_SCHEMA_VALIDATOR, self);
+        if !result.is_valid() {
+            return Ok(result);
+        }
+
+        //todo: re-enable version validation
+        // // Validate protocol version
+        // let protocol_version_validator = ProtocolVersionValidator::default();
+        // let result = protocol_version_validator
+        //     .validate(self.protocol_version)
+        //     .expect("TODO: again, how this will ever fail, why do we even need a validator trait");
+        // if !result.is_valid() {
+        //     return Ok(result);
+        // }
+        //
+        // // Validate data contract
+        // let data_contract_validator =
+        //     DataContractValidator::new(Arc::new(protocol_version_validator)); // ffs
+        // let result = data_contract_validator
+        //     .validate(&(self.data_contract.to_cleaned_object().expect("TODO")))?;
+        // if !result.is_valid() {
+        //     return Ok(result);
+        // }
+
+        // Validate data contract id
+        let generated_id =
+            generate_data_contract_id(self.data_contract.owner_id, self.data_contract.entropy);
+        if generated_id.as_slice() != self.data_contract.id.as_ref() {
+            return Ok(SimpleConsensusValidationResult::new_with_error(
+                BasicError::InvalidDataContractIdError(InvalidDataContractIdError::new(
+                    generated_id.to_vec(),
+                    self.data_contract.id.as_ref().to_owned(),
+                ))
+                .into(),
+            ));
+        }
+
+        self.data_contract
+            .validate_structure()
+            .map_err(Error::Protocol)
+    }
+
+    fn validate_identity_and_signatures(
+        &self,
+        drive: &Drive,
+        transaction: TransactionArg,
+    ) -> Result<ConsensusValidationResult<Option<PartialIdentity>>, Error> {
+        Ok(
+            validate_state_transition_identity_signature(drive, self, false, transaction)?
+                .map(Some),
+        )
+    }
+
+    fn validate_state<C: CoreRPCLike>(
+        &self,
+        platform: &PlatformRef<C>,
+        tx: TransactionArg,
+    ) -> Result<ConsensusValidationResult<StateTransitionAction>, Error> {
+        let drive = platform.drive;
+        // Data contract shouldn't exist
+        if drive
```

Comment:
So I'm wondering - why not keep the state repository interface, as before, and move these operations into it instead? That would be way more portable, and the interface can be just changed into a sync one if async doesn't work